pkg/cloudprovider/providers/openstack/openstack_loadbalancer.go

```diff
@@ -643,6 +654,100 @@ func (lbaas *CloudLb) EnsureLoadBalancer(ctx context.Context, clusterName string
 	return status, nil
 }
 
+func (lbaas *CloudLb) ensureLoadBalancerAccesslists(lbID uint64, sourceRangesCIDRs []accesslists.CreateOpts, serviceName string) error {
+	// Get the current access lists for the load balancer
+	accessLists, err := accesslists.Get(lbaas.lb, lbID).Extract()
+	if err != nil && !cpoerrors.IsNotFound(err) {
+		return fmt.Errorf("error getting access lists for load balancer %d: %v", lbID, err)
+	}
+	klog.V(2).Infof("Current access lists for load balancer %d: %v", lbID, accessLists)
+
+	sourceRangesCIDRSlices := []string{}
+	for _, sourceRange := range sourceRangesCIDRs {
+		sourceRangesCIDRSlices = append(sourceRangesCIDRSlices, sourceRange.Address)
+	}
+
+	// If the source ranges CIDRs are empty, we will delete all access lists
+	// i.e. if the source ranges CIDRs are equal to the default value
+	// v1service.DefaultLoadBalancerSourceRanges, we will delete all access lists.
+	// This is to ensure that if the source ranges CIDRs are not set, we do
+	// not leave any access lists that might have been set previously.
+	if len(sourceRangesCIDRSlices) == 1 && sourceRangesCIDRSlices[0] == v1service.DefaultLoadBalancerSourceRanges {
```

Comment:
Referring to this complete `if/else` block: As I understand it is finding the delta between the required and actual CIDR list. (and just deleting if only default range is specified)
How about we always delete all and then add what is needed to keep the logic simple?

Reply:
Rewritten the logic & added some conditions to avoid API errors.